cu_servicenow/cu_servicenow.module

```diff
@@ -0,0 +1,316 @@
+
+  $ch = curl_init(); // initialize curl handle
+  curl_setopt($ch, CURLOPT_HTTPHEADER, Array("Accept:application/json","Content-Type:application/json"));
+  curl_setopt($ch, CURLOPT_USERPWD, $conf['sn_key']);
+  curl_setopt($ch, CURLOPT_URL, $incident_assignment);
+  curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+  curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 10);
+  curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
+  $result = curl_exec($ch); // run the whole process
+  curl_setopt($ch, CURLOPT_HEADER, 1);
+  curl_setopt($ch, CURLOPT_VERBOSE, 1);
+  //$header = curl_exec($ch); // run the whole process
+  curl_close($ch); // close cURL handler
+
+  //$headers = _get_headers_from_curl_response($header);
+  $result = json_decode($result);
+
+  return $result;
+}
+
+/**
+ * ServiceNow API call via cURL.
+ *
+ * Send in form_values from the custom submit handler.
+ * The values must match available fields in ServiceNow.
+ */
+function _servicenow_curl_request($form_values) {
+  global $conf;
+
+  $request = variable_get('sn_environment', 'https://coloradosandbox.service-now.com') . '/api/now/table/incident';
+  $content_encode = json_encode($form_values);
+
+  $user_info = $conf['sn_key'];
+
+  $ch = curl_init(); // initialize curl handle
+  curl_setopt($ch, CURLOPT_HTTPHEADER, Array("Accept:application/json","Content-Type:application/json"));
+  curl_setopt($ch, CURLOPT_USERPWD, $user_info);
+  curl_setopt($ch, CURLOPT_URL, $request);
+  curl_setopt($ch, CURLOPT_POSTFIELDS, $content_encode);
+  curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+  curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 10);
+  curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, false);
+  $result = curl_exec($ch); // run the whole process
+  curl_setopt($ch, CURLOPT_HEADER, 1);
+  curl_setopt($ch, CURLOPT_VERBOSE, 1);
+  //$header = curl_exec($ch); // run the whole process
+  curl_close($ch); // close cURL handler
+
+  //$headers = _get_headers_from_curl_response($header);
+  $result = json_decode($result);
+
+  //$result->headers = $headers;
+  if ( (!isset($result->status)) && (!empty($result)) ) {
+    $result->status = 'success';
+  } else {
+    $result->status = 'failure';
+  }
+  if ( ($result->status == 'failure') || (empty($result)) ) {
+    watchdog('cu_servicenow', 'servicenow api not communicating correctly \_(ツ)_/¯ api call: ' . $request);
+    drupal_set_message(t('There was a problem sending your request to ServiceNow. Please contact the site owner.'), 'success');
+    return null;
+  } else {
+    $inc_num = $result->result->number;
+    drupal_set_message(t('Your incident number is :incident, please keep this number for your records.', array(':incident' => $inc_num)), 'success');
+    return $result;
```

Comment:
Be careful on return types where this function is called and what they expect to be returned.